### PR TITLE
Toolbar: Bold, Italic and Underline buttons

### PR DESCRIPTION
**Pull request by kradchenko**
## What?
I've added to the toolbar the following new buttons: `Bold`, `Italic`, and `Underline`. These buttons work as toggle buttons for their respective functions.

## Why?
So users can create interesting thought trees and highlight thoughts that they decide as important.

## How?
I've created new `Shortcuts` instances for the `Bold`, `Italic`, and `Underline` buttons. I've created the necessary reducer functions for each button. Also, I've added keyboard shortcuts for each button.

## Anything else?
I couldn't test the code in the mobile version of the application as I got a `compilation error` while running the `yarn run start:expo`.

### CODE REVIEW

File: src/components/icons/BoldTextIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import Icon from '../../@types/Icon'
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const BoldTextIcon = ({ style, size }: Icon) => {
```
Comment (raineorshine) [inline src/components/icons/BoldTextIcon.tsx:5]: I'd like an icon without a border with a more bold B.

File: src/components/icons/ItalicTextIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import Icon from '../../@types/Icon'
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const ItalicTextIcon = ({ style, size }: Icon) => {
```
Comment (raineorshine) [inline src/components/icons/ItalicTextIcon.tsx:5]: Let's remove the border on this icon.

File: src/components/icons/UnderlineIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import Icon from '../../@types/Icon'
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+const UnderlineIcon = ({ size = 20, style }: Icon) => (
```
Comment (raineorshine) [inline src/components/icons/UnderlineIcon.tsx:5]: Currently the "U" is much larger than the "B" and "I". The "U" should match the size of "B" and "I" for visual consistency.

File: src/reducers/toggleTextBoldness.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { toggleAttribute } from '.'
+import _ from 'lodash'
+import State from '../@types/State'
+
+/** Toggles the bold font weight of the cursor. */
+const toggleTextBoldness = (state: State) => {
+  if (!state.cursor) return state
+  const path = state.cursor
+
+  return toggleAttribute(state, { path, values: ['=style', 'fontWeight', '700'] })
```
Comment (raineorshine) [inline src/reducers/toggleTextBoldness.ts:10]: Looks good!

File: src/shortcuts/toggleTextBoldness.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import Shortcut from '../@types/Shortcut'
+import Icon from '../components/icons/BoldTextIcon'
+import isDocumentEditable from '../util/isDocumentEditable'
+
+/** Toolbars shortcut instance of toggleTextBoldness, which toggles bold font weight of the cursor. */
+const toggleTextBoldness: Shortcut = {
+  id: 'toggleTextBoldness',
+  label: 'Boldness Toggle',
```
Comment (raineorshine) [inline src/shortcuts/toggleTextBoldness.tsx:8]: Label should be "Bold"

File: src/shortcuts/toggleTextItalicStyle.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import Shortcut from '../@types/Shortcut'
+import Icon from '../components/icons/ItalicTextIcon'
+import isDocumentEditable from '../util/isDocumentEditable'
+
+/** Toolbars shortcut instance of toggleTextItalicStyle, which toggles italic font style of the cursor. */
+const toggleTextItalicStyle: Shortcut = {
+  id: 'toggleTextItalicStyle',
+  label: 'Italic Toggle',
```
Comment (raineorshine) [inline src/shortcuts/toggleTextItalicStyle.tsx:8]: Label should be "Italic".

File: src/shortcuts/toggleTextUnderline.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import Shortcut from '../@types/Shortcut'
+import Icon from '../components/icons/UnderlineIcon'
+import isDocumentEditable from '../util/isDocumentEditable'
+
+/** Toolbars shortcut instance of toggleTextUnderline, which toggles underline decoration of the cursor. */
+const toggleTextUnderline: Shortcut = {
+  id: 'toggleTextUnderline',
+  label: 'Underline Toggle',
```
Comment (raineorshine) [inline src/shortcuts/toggleTextUnderline.tsx:8]: Label should be "Underline".

File: src/constants.ts [hunk 8]
```diff
@@ -228,6 +228,9 @@ export const TOOLBAR_DEFAULT_SHORTCUTS: (keyof typeof shortcuts)[] = [
   'splitSentences',
   'toggleHiddenThoughts',
   'exportContext',
+  'toggleTextBoldness',
+  'toggleTextItalicStyle',
+  'toggleTextUnderline',
```
Comment (raineorshine) [inline src/constants.ts:233]: Let's move these after `textColor`.

**Review by raineorshine — CHANGES_REQUESTED** [review]
Looks great! Thanks for following the conventions of the project.

Just a few small requested changes.

File: src/components/icons/BoldTextIcon.tsx (hunk 1 above)
Comment (kradchenko) [reply src/components/icons/BoldTextIcon.tsx:5]: I've changed Icon. Right now it's without a border and B is bolder.

File: src/components/icons/ItalicTextIcon.tsx (hunk 2 above)
Comment (kradchenko) [reply src/components/icons/ItalicTextIcon.tsx:5]: I've changed Icon. The border was removed and _I_ is bigger.

File: src/components/icons/UnderlineIcon.tsx (hunk 3 above)
Comment (kradchenko) [reply src/components/icons/UnderlineIcon.tsx]: I've changed the UnderlineIcon. I tried to match the "B", "I" and "U" icons.

File: src/shortcuts/toggleTextBoldness.tsx (hunk 5 above)
Comment (kradchenko) [reply src/shortcuts/toggleTextBoldness.tsx]: I've changed label to "Bold".

File: src/shortcuts/toggleTextItalicStyle.tsx (hunk 6 above)
Comment (kradchenko) [reply src/shortcuts/toggleTextItalicStyle.tsx]: I've changed the label to "Italic".

File: src/shortcuts/toggleTextUnderline.tsx (hunk 7 above)
Comment (kradchenko) [reply src/shortcuts/toggleTextUnderline.tsx]: I've changed the label to "Underline".

File: src/constants.ts (hunk 8 above)
Comment (kradchenko) [reply src/constants.ts]: I've moved these toggles after textColor.

**Comment by kradchenko** [comment]
I've addressed all comments. Please, take a look.

cc @raineorshine

**Review by raineorshine — CHANGES_REQUESTED** [review]
Thanks, looks good.

The only issue I see is that the icons do not scale with the font size. You can adjust the font size using the links in the bottom left of the footer. Take a look at how the other icons scale and make sure this functionality is preserved.

For example, here is the what the Toolbar looks like at Font Size 13:

image

Please make sure the size and alignment looks correct at all font sizes.

**Comment by kradchenko** [comment]
I've made Icons responsive. Right now, I think they look good in all font sizes.

cc @raineorshine

**Review by raineorshine — CHANGES_REQUESTED** [review]
Reviewing now

**Comment by kradchenko** [comment]
I get it! I will look into it.

**Review by raineorshine — APPROVED** [review]
Looks good! 

I made a few final tweaks to get the size and positioning right, changed the toolbar icon highlight logic, and renamed the files.

**Pull request merged**